app/src/main/java/com/jwisozk/flightreviews/SubmitFragment.kt

```diff
@@ -0,0 +1,93 @@
+        super.onViewCreated(view, savedInstanceState)
+        val activity = requireActivity() as MainActivity
+        val rootLayout: CoordinatorLayout = view.findViewById(R.id.submitFragment)
+        val spinner: ProgressBar = view.findViewById(R.id.spinner)
+        val repository = Repository(getNetworkService())
+        val param = Parameters()
+        val viewModel = ViewModelProviders
+            .of(activity, ParamViewModel.FACTORY(repository))
+            .get(ParamViewModel::class.java)
+        val viewManager = LinearLayoutManager(activity)
+        viewModel.setParameters(param)
+        val ratingBar = view.findViewById<RatingBar>(R.id.appbarRatingBar)
+        ratingBar.onRatingBarChangeListener =
+            RatingBar.OnRatingBarChangeListener { _, rating, _ ->
+                viewModel.onOverallRatingChanged(rating, ParamEnum.FLIGHT)
+            }
+        val motionLayout = view.findViewById<MotionLayout>(R.id.motionLayout)
+        val appBar = view.findViewById<AppBarLayout>(R.id.appBarLayout)
+        appBar.addOnOffsetChangedListener(object: MotionLayout(activity), AppBarLayout.OnOffsetChangedListener {
+            override fun onOffsetChanged(appBarLayout: AppBarLayout?, verticalOffset: Int) {
+                appBarLayout?.totalScrollRange?.toFloat()?.let {
+                    motionLayout.progress = -verticalOffset / it
+                }
+            }
+            override fun onAttachedToWindow() {
+                super.onAttachedToWindow()
+                (parent as? AppBarLayout)?.addOnOffsetChangedListener(this)
+            }
+        })
+
+        val viewAdapter = ReviewAdapter(viewModel)
+        view.findViewById<RecyclerView>(R.id.recycler).apply {
+            setHasFixedSize(true)
+            layoutManager = viewManager
+            adapter = viewAdapter
+        }
+
+        // show the isLoading when [ParamViewModel.isLoading] is true
+        viewModel.isLoading.observe(viewLifecycleOwner) { value ->
+            value?.let { show ->
+                if (show) {
+                    spinner.visibility = View.VISIBLE
+                    val button = view.findViewById<Button>(R.id.submitButton)
+                    button.visibility = View.GONE
+                } else if (spinner.visibility == View.VISIBLE) {
+                    spinner.visibility = View.GONE
+                    val button = view.findViewById<Button>(R.id.submitButton)
+                    button.visibility = View.VISIBLE
+                }
+            }
+        }
+
+        // Show a snackbar whenever the [ParamViewModel.snackbar] is updated a non-null value
+        viewModel.snackbar.observe(viewLifecycleOwner) { text ->
+            text?.let {
+                Snackbar.make(rootLayout, text, Snackbar.LENGTH_SHORT).show()
+            }
+        }
+
+        // Transition to SuccessFragment
+        viewModel.isRefreshData.observe(viewLifecycleOwner) { value ->
+            value?.let { isNextFragment ->
```

Comment:
`let` is just syntax sugar, as I understand it creates an object upon invocation, so there is no point using it here instead it sould use early return, something like this:
```kotlin
if (value != true) {
    // goes here if null as well
    return@observe
}
activity.transitionToSuccessFragment()
```
this is much more efficient and less confusing for others, same for all other places btw